Mathlib/Algebra/Lie/BaseChange.lean

```diff
@@ -26,36 +26,35 @@ lie ring, lie algebra, extension of scalars, restriction of scalars, base change
 
 suppress_compilation
 
-universe u v w w₁ w₂ w₃
-
 open scoped TensorProduct
 
-variable (R : Type u) (A : Type w) (L : Type v)
+variable (R A L M : Type*)
 
 namespace LieAlgebra
 
 namespace ExtendScalars
 
 variable [CommRing R] [CommRing A] [Algebra R A] [LieRing L] [LieAlgebra R L]
+  [AddCommGroup M] [Module R M] [LieRingModule L M] [LieModule R L M]
 
 /-- The Lie bracket on the extension of a Lie algebra `L` over `R` by an algebra `A` over `R`. -/
-private def bracket' : A ⊗[R] L →ₗ[A] A ⊗[R] L →ₗ[A] A ⊗[R] L :=
+private def bracket' : A ⊗[R] L →ₗ[A] A ⊗[R] M →ₗ[A] A ⊗[R] M :=
   TensorProduct.curry <|
     TensorProduct.AlgebraTensorModule.map
-        (LinearMap.mul' A A) (LieModule.toModuleHom R L L : L ⊗[R] L →ₗ[R] L) ∘ₗ
-      (TensorProduct.AlgebraTensorModule.tensorTensorTensorComm R A A L A L).toLinearMap
+        (LinearMap.mul' A A) (LieModule.toModuleHom R L M : L ⊗[R] M →ₗ[R] M) ∘ₗ
+      (TensorProduct.AlgebraTensorModule.tensorTensorTensorComm R A A L A M).toLinearMap
 
 @[simp]
-private theorem bracket'_tmul (s t : A) (x y : L) :
-    bracket' R A L (s ⊗ₜ[R] x) (t ⊗ₜ[R] y) = (s * t) ⊗ₜ ⁅x, y⁆ := rfl
+private theorem bracket'_tmul (s t : A) (x : L) (m : M) :
+    bracket' R A L M (s ⊗ₜ[R] x) (t ⊗ₜ[R] m) = (s * t) ⊗ₜ ⁅x, m⁆ := rfl
 
-instance : Bracket (A ⊗[R] L) (A ⊗[R] L) where bracket x y := bracket' R A L x y
+instance : Bracket (A ⊗[R] L) (A ⊗[R] M) where bracket x m := bracket' R A L M x m
```

Comment:
I worry slightly that there's potential for an instance diamond here (generally instances of the form `[Foo A B] : Foo (Structure A) (Structure B)` are risky if there's any chance that `B = Structure A`), but I'm happy to let that be a problem for future Oliver if it ever arises.

Reply:
I think we're safe but in any case I agree.